externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -191,4 +191,100 @@ def Tcp_BroadcastOp : Tcp_Op<"broadcast", [
   let hasVerifier = 1;
 }
 
+def Tcp_YieldOp : Tcp_Op<"yield", [Terminator, Pure]> {
+  let summary = "Yields values from regions in Tcp";
+
+  let description = [{
+    Terminates and yields values from regions in Tcp.
+
+    The values that are yielded should correspond to the parent operation's
+    results.
+  }];
+
+  let arguments = (ins
+    Variadic<Tcp_TensorOrScalar>:$ins
+  );
+
+  let builders = [
+    OpBuilder<(ins), [{ }]>,
+  ];
+
+  let assemblyFormat = "$ins attr-dict `:` type($ins)";
+}
+
+def Tcp_GroupOp : Tcp_Op<"group", [
+          AffineScope,
+          AutomaticAllocationScope,
+          ParentOneOf<["func::FuncOp", "tcp::GroupOp", "tcp::IsolatedGroupOp"]>,
+          SingleBlockImplicitTerminator<"tcp::YieldOp">]> {
+  let summary = "Groups ops into a region that is not isolated from above";
+
+  let description = [{
+    Provides a way to group ops into a region that is not isolated from above.
+
+    Since this is not isolated from above, ops inside this region can refer to
+    values in the outer scopes. So, this op does not have any input parameters.
+
+    The group_type attribute is a free-form string that can be used to denote
+    any aspect of the group. For example, it could be set to "elementwise" for
+    a group that only contains elementwise ops.
```

Comment:
Please note here what semantic effect the `group_type` string has, if any.  E.g. if it is `"elementwise"` then does that mean the group body cannot have non-elementwise ops?

My suggestion: I suggest we say that `group_type` is purely a hint and has no semantic meaning.  So even if it is `"elementwise"`, a downstream pass cannot rely on this information for correctness and must explicitly check the body of the group op.  Maybe it should be called `group_hint` actually.

Reply:
I would completely drop it from the op definition. It seems too vague and free form and is just a StrAttr. Maybe with the properties proposal, such things won't be necessary. Clients can add properties to ops to "classify" groups

Reply:
Good points regarding `group_type`. As outlined in the RFC doc, we are not clear how this is going to be used. In this situation I agree that having such a free-from string attribute may make things more confusing / erroneous. So, removed the attribute now.

If properties are meant to handle such cases, that would be great. We can look to use that once it is available.